### PR TITLE
DERP-631 Manage related items component

### PR DESCRIPTION
**Pull request by VincenzoGambino**
Related ticket: https://london.atlassian.net/browse/DERP-631
Related PR: https://github.com/GreaterLondonAuthority/lgov/pull/88

### CODE REVIEW

File: modules/gla_core_related_items/gla_core_related_items.module [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+
+/**
+ * Allowed values callback for field_cl_content_type.
+ *
+ * @return array
+ *   Content type option list.
+ */
+function gla_core_related_items_content_types() {
+  $node_types = NodeType::loadMultiple();
+  // TODO: Make this configurable in Voyager.
```
[the remark this answers is not in the document]
Comment (joz-cti) [reply modules/gla_core_related_items/gla_core_related_items.module:70]: A comment on the purpose of this function would be helpful as I wouldn't know where this is being used in 6 months time.

[the remark this answers is not in the document]
Comment (joz-cti) [reply modules/gla_core_related_items/gla_core_related_items.module:70]: Are these being used for a custom service which is later used in AutomatedRelatedItems.php ?
A comment like the above would provide context.

File: modules/gla_core_related_items/src/Plugin/RelatedItemsType/AutomatedRelatedItems.php [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+    }
+    if (!empty($tags)) {
+      $query->condition('field_tags', $tags, 'IN');
+    }
+    if ($limit) {
+      $query->pager($limit);
+    }
+    $query->sort('published_at', 'DESC');
+    $query->sort('created', 'DESC');
+    $query->latestRevision();
```
Comment (joz-cti) [inline modules/gla_core_related_items/src/Plugin/RelatedItemsType/AutomatedRelatedItems.php:79]: Is this still being used? If we have gone for a Views approach?

**Review by joz-cti — DISMISSED** [review]

File: modules/gla_core_related_items/gla_core_related_items.module [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+    $build['field_p_ri_content'] = $entity->field_p_ri_content->view($display->getComponent('field_p_ri_content'));
+
+  }
+}
+
+/**
+ * Implements hook_field_widget_form_alter().
+ */
+function gla_core_related_items_field_widget_paragraphs_form_alter(&$element, FormStateInterface $form_state, $context) {
+  if ($element['#paragraph_type'] == 'related_items') {
```
[the remark this answers is not in the document]
Comment (VincenzoGambino) [reply modules/gla_core_related_items/gla_core_related_items.module:41]: Added comment.

File: modules/gla_core_related_items/src/Plugin/RelatedItemsType/AutomatedRelatedItems.php (hunk 2 above)
Comment (VincenzoGambino) [reply modules/gla_core_related_items/src/Plugin/RelatedItemsType/AutomatedRelatedItems.php:81]: We are altering the EntityViewDisplay

File: modules/gla_core_related_items/gla_core_related_items.module (hunk 1 above)
[the remark this answers is not in the document]
Comment (VincenzoGambino) [reply modules/gla_core_related_items/gla_core_related_items.module]: This function is used in the field configuration.  I have added the comment.

File: modules/gla_core_related_items/gla_core_related_items.module [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+use Drupal\Core\Entity\EntityInterface;
+use Drupal\Core\Entity\Display\EntityViewDisplayInterface;
+use Drupal\node\Entity\NodeType;
+
+/**
+ * Implements hook_entity_view_alter().
+ */
+function gla_core_related_items_paragraph_view_alter(array &$build, EntityInterface $entity, EntityViewDisplayInterface $display) {
+  if ($entity->bundle() == 'related_items') {
+    $type = $entity->get('field_p_ri_listing_type')->getValue();
```
[the remark this answers is not in the document]
Comment (VincenzoGambino) [reply modules/gla_core_related_items/gla_core_related_items.module]: Done.

**Review by Fullstack-Lincoln — APPROVED** [review]

**Pull request merged**